extensions/pkg/predicate/predicate_test.go

```diff
@@ -87,6 +90,97 @@ var _ = Describe("Predicate", func() {
 		})
 	})
 
+	Describe("#AddTypePredicate", func() {
+		var (
+			object       client.Object
+			createEvent  event.CreateEvent
+			updateEvent  event.UpdateEvent
+			deleteEvent  event.DeleteEvent
+			genericEvent event.GenericEvent
+		)
+
+		BeforeEach(func() {
+			object = &extensionsv1alpha1.Extension{
+				Spec: extensionsv1alpha1.ExtensionSpec{
+					DefaultSpec: extensionsv1alpha1.DefaultSpec{
+						Type: extensionType,
+					},
+				},
+			}
+			createEvent = event.CreateEvent{
+				Object: object,
+			}
+			updateEvent = event.UpdateEvent{
+				ObjectOld: object,
+				ObjectNew: object,
+			}
+			deleteEvent = event.DeleteEvent{
+				Object: object,
+			}
+			genericEvent = event.GenericEvent{
+				Object: object,
+			}
+		})
+
+		It("should return true because only one extension exists and extension type matches", func() {
+			predicates := AddTypePredicate([]predicate.Predicate{HasType("foo")}, extensionType)
+
+			Expect(len(predicates)).To(Equal(2))
+
+			Expect(predicates[1].Create(createEvent)).To(BeTrue())
+			Expect(predicates[1].Update(updateEvent)).To(BeTrue())
+			Expect(predicates[1].Delete(deleteEvent)).To(BeTrue())
+			Expect(predicates[1].Generic(genericEvent)).To(BeTrue())
+		})
+
+		It("or predicate should return true because one of extension type match", func() {
+			predicates := AddTypePredicate([]predicate.Predicate{HasType("foo")}, extensionType, "bar")
```

Comment:
Will it make much difference by adding `HasPurpose` instead of `HasType`. This test case anyway checks whether predicate is added or not. Be it `HasPurpose` or `HasType`, it doesn't make a difference.

Reply:
I don't have any strong opinions. But I feel it will look better with different predicates which is more closer to actual use case.